crates/sui-framework/sources/collectible/kiosk.move

```diff
@@ -0,0 +1,474 @@
+    const EIncorrectAmount: u64 = 2;
+    /// For when incorrect arguments passed into `switch_mode` function.
+    const EIncorrectArgument: u64 = 3;
+    /// For when Transfer is accepted by a wrong Kiosk.
+    const EWrongTarget: u64 = 4;
+    /// For when trying to withdraw higher amount than stored.
+    const ENotEnough: u64 = 5;
+    /// For when trying to close a Kiosk and it has items in it.
+    const ENotEmpty: u64 = 6;
+
+    /// An object that stores collectibles of all sorts.
+    /// For sale, for collecting reasons, for fun.
+    struct Kiosk has key, store {
+        id: UID,
+        /// Balance of the Kiosk - all profits from sales go here.
+        profits: Balance<SUI>,
+        /// Always point to `sender` of the transaction.
+        /// Can be changed by calling `set_owner` with Cap.
+        owner: address,
+        /// Number of items stored in a Kiosk. Used to allow unpacking
+        /// an empty Kiosk if it was wrapped or has a single owner.
+        item_count: u32
+    }
+
+    /// A capability that is issued for Kiosks that don't have owner
+    /// specified.
+    struct KioskOwnerCap has key, store {
+        id: UID,
+        for: ID
+    }
+
+    /// A "Hot Potato" forcing the buyer to get a transfer permission
+    /// from the item type (`T`) owner on purchase attempt.
+    struct TransferRequest<phantom T: key + store> {
+        /// Amount of SUI paid for the item. Can be used to
+        /// calculate the fee / transfer policy enforcement.
+        paid: u64,
+        /// The ID of the Kiosk the object is being sold from.
+        /// Can be used by the TransferPolicy implementors to
+        /// create an allowlist of Kiosks which can trade the type.
+        from: ID,
+    }
+
+    /// A unique capability that allows owner of the `T` to authorize
+    /// transfers. Can only be created with the `Publisher` object.
+    struct TransferPolicyCap<phantom T: key + store> has key, store {
+        id: UID
+    }
+
+    // === Dynamic Field keys ===
+
+    /// Dynamic field key for an item placed into the kiosk.
+    struct Key has store, copy, drop { id: ID }
+
+    /// Dynamic field key for an active offer to purchase the T.
+    struct Offer has store, copy, drop { id: ID }
+
+    // === Events ===
+
+    /// Emitted when an item was listed by the safe owner. Can be used
+    /// to track available offers anywhere on the network; the event is
+    /// type-indexed which allows for searching for offers of a specific `T`
+    struct NewOfferEvent<phantom T: key + store> has copy, drop {
+        kiosk: ID,
+        id: ID,
+        price: u64
+    }
+
+    /// Emitted when a publisher creates a new `TransferPolicyCap` making
+    /// the discoverability and tracking the supported types easier.
+    struct TransferPolicyCapIssued<phantom T: key + store> has copy, drop {
+        id: ID
+    }
+
+    // === New Kiosk + ownership modes ===
+
+    /// Creates a new Kiosk without owner but with a Capability.
```

Comment:
What's the distinction between the owner of the `KioskOwnerCap` and the address in `kiosk.owner`?  I'm a little bit confused by this doc comment because it says the `Kiosk` doesn't have an `owner` but we see that its `owner` field is set to the transaction sender.

Taking this further, should the `owner` of the `Kiosk` be the `ID` of its cap instead?